libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts

```diff
@@ -83,10 +83,11 @@ export interface IElementsQueryFactory {
      * Query attribute elements used by provided filter
      *
      * @param filter - resolvable filter
+     * @param dateFilterDisplayForm - display form of resolvable filter if it is date filter
      * @returns instance that can be used to query attribute elements
      *
      */
-    forFilter(filter: FilterWithResolvableElements): IFilterElementsQuery;
+    forFilter(filter: FilterWithResolvableElements, dateFilterDisplayForm?: ObjRef): IFilterElementsQuery;
```

Comment:
Idea: can we somehow use function overloads so that the second parameter is only there if the first parameter is a date filter? See [Ranking filter factory](https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-model/src/execution/filter/factory.ts#L281-L362) for inspiration.

Reply:
This approach cant be used on methods of interface, only pure functions
https://github.com/microsoft/TypeScript/issues/7230